### PR TITLE
Small decode sorries

### PR DESCRIPTION
**Pull request by michaelmcinerney**
Signed-off-by: Michael McInerney <Michael.McInerney@data61.csiro.au>

### CODE REVIEW

File: proof/refine/ARM/Tcb_R.thy [hunk 1]
```diff
@@ -3143,6 +3143,13 @@ lemma checkPrio_lt_ct_weak:
   apply (clarsimp simp: pred_tcb_at'_def obj_at'_def)
   by (rule le_ucast_ucast_le) simp
 
+lemma checkPrio_lt_ct_weak':
+  "\<lbrace>P\<rbrace> checkPrio prio auth \<lbrace>\<lambda>rv s. P s \<and> mcpriority_tcb_at' (\<lambda>mcp. ucast prio \<le> mcp) auth s\<rbrace>, -"
+  apply (wpsimp wp: hoare_vcg_conj_liftE1)
+    apply (wpsimp wp: checkPrio_inv)
+   apply (wpsimp wp: checkPrio_lt_ct_weak)+
+  done
```
Comment (MitchellBuckley) [inline proof/refine/ARM/Tcb_R.thy:3151]: nitpick: merge into one wpsimp if possible (but I know it's not always possible).

**Review by MitchellBuckley — APPROVED** [review]
Noice. You're a machine!

File: proof/refine/ARM/Tcb_R.thy (hunk 1 above)
Comment (michaelmcinerney) [reply proof/refine/ARM/Tcb_R.thy:3151]: No dice, unfortunately

**Review by mktnk3 — APPROVED** [review]
Nice. This will put us below 80 sorries :)

File: proof/refine/ARM/Tcb_R.thy [hunk 2]
```diff
@@ -3143,6 +3143,13 @@ lemma checkPrio_lt_ct_weak:
   apply (clarsimp simp: pred_tcb_at'_def obj_at'_def)
   by (rule le_ucast_ucast_le) simp
 
+lemma checkPrio_lt_ct_weak':
+  "\<lbrace>P\<rbrace> checkPrio prio auth \<lbrace>\<lambda>rv s. P s \<and> mcpriority_tcb_at' (\<lambda>mcp. ucast prio \<le> mcp) auth s\<rbrace>, -"
```
Comment (corlewis) [inline proof/refine/ARM/Tcb_R.thy:3147]: Ooh, that's a nice way of solving the problem we discussed. I guess that means that the idea of replacing `checkPrio_inv` with several more specific lemmas didn't work though?

**Review by corlewis — APPROVED** [review]

File: proof/refine/ARM/Tcb_R.thy (hunk 2 above)
Comment (michaelmcinerney) [reply proof/refine/ARM/Tcb_R.thy:3147]: Yeah that quickly became annoying. The `wp` produced some postconditions where the preemption case was not simply `True`, and I thought the approach to fixing it was worse than the slightly more manual forwards proof. This proof does take quite a while to get through the case spliting in the first few lines, but I think it's much clearer what's going on

**Pull request merged**